NewHaven/test/Part1Driver.cpp

```diff
@@ -0,0 +1,11 @@
+//
+// Created by hyperion on 2020-03-17.
+//
+
+#include "Setting.h"
+
+int main(){
+    Setting setting;
+    setting.current->initSetting();
+
+};
```

Comment:
Many of your functions throw exceptions, but you don't handle anything here.

Reply:
They do, but initSetting is a controlled environment. Unless someone removes a file from a directory no exception will ever be flagged with initSetting.